src/Microsoft.AspNetCore.Razor.Evolution/DefaultDirectiveIRPass.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Evolution.Intermediate;
+using Microsoft.AspNetCore.Razor.Evolution.Legacy;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    internal class DefaultDirectiveIRPass : IRazorIRPass
+    {
+        public RazorEngine Engine { get; set; }
+
+        public int Order => 150;
+
+        public DocumentIRNode Execute(RazorCodeDocument codeDocument, DocumentIRNode irDocument)
+        {
+            var walker = new DirectiveWalker();
+            walker.VisitDefault(irDocument);
+
+            return irDocument;
+        }
+
+        private class DirectiveWalker : RazorIRNodeWalker
+        {
+            private ClassDeclarationIRNode _classNode;
+
+            public override void VisitClass(ClassDeclarationIRNode node)
+            {
+                if (_classNode == null)
+                {
+                    _classNode = node;
+                }
+
+                VisitDefault(node);
+            }
+
+            public override void VisitDirective(DirectiveIRNode node)
+            {
+                if (string.Equals(node.Name, CSharpCodeParser.FunctionsDirectiveDescriptor.Name, StringComparison.Ordinal))
+                {
+                    foreach (var child in node.Children.Except(node.Tokens))
```

Comment:
So I'm not removing the functions directive from where it originally was because it needs to have some presence as a `DirectiveIRNode` so it can be consumed at `DesignTime` to provide proper IntelliSense.

Options:
1. Do what we're doing now, let it exist in two places. This means we have to choose not to `VisitDefault` when we code gen.
2. Remove from razor method declaration and extract its token ir nodes and the functions body to the class level. I believe we can make due with just the tokens to provide IntelliSense.
3. Move the entire directive node to the class level and have code gen special case the functions directive.
4. `DirectiveIRNodes` are not allowed to make it to the code gen phase. There's a new pass that extracts all tokens into a special location and nukes the DirectiveIRNodes from the IR tree. This approach assumes IR passes have transformed directives into their final form and left behind artifacts.

I'm leaning towards 4 but it's more work.